account_payment_term_partner_holiday/models/account_invoice.py

```diff
@@ -0,0 +1,28 @@
+# Copyright 2021 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models
+
+
+class AccountInvoice(models.Model):
+    _inherit = "account.invoice"
+
+    @api.onchange("date_due")
+    def _onchange_date_due(self):
+        if self.date_due and self.payment_term_id and self.partner_id:
```

Comment:
This is not the goal of the module (although it can be covered as well), but to modify the date when using payment terms, that make use of this method:

https://github.com/odoo/odoo/blob/3353fef07a3a54596b30aabc29d9729f2eb3875e/addons/account/models/account_invoice.py#L2057

The best technique is to call super, and the goes through each returned line for modifying forbidden dates.